src/nbody/nbody_utils.hpp

```diff
@@ -34,12 +34,30 @@ namespace NBody {
 extern void reb_extra_forces(struct reb_simulation *rsim);
 extern int collision_resolution(struct reb_simulation *const r, struct reb_collision c);
 
+// Container for reb_simulation pointer that allows it to be stored in Params with
+// automatic cleanup
+struct RebSim {
+
+  struct reb_simulation *&get() {
```

Comment:
Can we add several constructors to this as well?
```
RebSim() {
 reb_sim = reb_simulation_create();
}
RebSim(struct rebound_simulation *r_sim) {
  reb_sim = reb_simulation_copy(r_sim);
}
RebSim(std::string filename) {
  reb_sim = reb_simulation_create_from_file(filename);
}
```

Reply:
1) Having the default constructor will do extra work of creating a rebound simulation that we discard for this case:
```c++
  RebSim reb_sim_rst;
  reb_sim_rst = nbody_pkg->Param<RebSim>("reb_sim");
```
so I don't think we want that unless I'm missing something
2) I added a custom copy constructor to avoid `.get()`s in the code
3) I added the std::string filename constructor
:+1:

Reply:
Could you make a constructor that allows `RebSim reb_sim_rst(nbody_pkg->Param<RebSim>("reb_sim")`. 

The reason for 1 is just to have a true RAII class. Feels wrong to acquire the resource outside of the constructor.

Reply:
OK actually yeah I can just use 2) for the case you described. I did that and made the default constructor do a `reb_simulation_create`, I agree this is nicest.

Reply:
OK having a `reb_simulation_copy` inside the copy constructor appears to break the code during internal `Params` manipulations. I reverted to default copy constructor and added a `RebSim::copy` method, which is also a bit more explicit about what we're doing.